src/sw.ts

```diff
@@ -91,23 +91,34 @@ const isRootRequestForContent = (event: FetchEvent): boolean => {
   return isRootRequest // && getCidFromUrl(event.request.url) != null
 }
 
-function getSubdomainParts (request: Request): { origin: string | null, protocol: string | null } {
+function getSubdomainParts (request: Request): { id: string | null, protocol: string | null } {
   const urlString = request.url
-  const url = new URL(urlString)
-  const subdomain = url.hostname.replace(`.${BASE_URL}`, '')
-  const subdomainRegex = /^(?<origin>[^/]+)\.(?<protocol>ip[fn]s)?$/
```

Comment:
:information_source:  this produced invalid results for `docs.ipfs.tech.ipns.example.com`, the new code walks labels from right to left, which is better order for avoiding false-positive matches.

Reply:
should we redirect from subdomains like "docs.ipfs.tech" to "docs-ipfs-tech" ?